dotCMS/src/main/java/com/dotcms/publisher/pusher/AuthCredentialPushPublishUtil.java

```diff
@@ -0,0 +1,205 @@
+package com.dotcms.publisher.pusher;
+
+import com.dotcms.auth.providers.jwt.JsonWebTokenAuthCredentialProcessor;
+import com.dotcms.auth.providers.jwt.beans.JWToken;
+import com.dotcms.auth.providers.jwt.services.JsonWebTokenAuthCredentialProcessorImpl;
+import com.dotcms.publisher.endpoint.bean.PublishingEndPoint;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.cms.factories.PublicEncryptionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Config;
+import com.liferay.portal.model.User;
+import io.jsonwebtoken.*;
+import org.apache.commons.lang.StringUtils;
+import org.glassfish.jersey.server.ContainerRequest;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * Singleton class to provide util method to Push Publish authentication, it support two authentication methods:
+ * JWToken and Auth key set in the {@link com.dotcms.publisher.endpoint.bean.impl.PushPublishingEndPoint},
+ * if the config property 'USE_JWT_TOKEN_IN_PUSH_PUBLISH' is set to true (default value)
+ * it is going to use JWT token, but if the config property is set to false or the JWT token fails then auth key is going to use
+ * */
+public enum AuthCredentialPushPublishUtil {
+    INSTANCE;
+
+    private final String BEARER = "Bearer ";
```

Comment:
Codacy found an issue: [This final field could be made static](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)